src/components/ProgramRecord/ProgramRecordActions.jsx

```diff
@@ -0,0 +1,237 @@
+import React, { useState, useEffect } from 'react';
+
+import PropTypes from 'prop-types';
+import {
+  Help, AccountBalance, ContentCopy, Download, SpinnerSimple, Check,
+} from '@edx/paragon/icons';
+import {
+  Button, StatefulButton, Icon, IconButton, OverlayTrigger, Popover, Tooltip, Toast,
+} from '@edx/paragon';
+
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { logError } from '@edx/frontend-platform/logging';
+
+import { getConfig } from '@edx/frontend-platform';
+import { getProgramRecordUrl, getProgramRecordCsv } from './data/service';
+
+const ProgramRecordActions = ({
+  showSendRecordButton, isPublic, toggleSendRecordModal, renderBackButton, username, programId,
+}) => {
+  const [programRecordUrl, setProgramRecordUrl] = useState('');
+  const [showCopyTooltip, setShowCopyTooltip] = useState(false);
+  const [showDownloadToast, setShowDownloadToast] = useState(false);
+  const [downloadRecord, setDownloadRecord] = useState('default');
+
+  const downloadButtonProps = {
+    labels: {
+      default: 'Download program record',
+      pending: 'Downloading program record',
+      complete: 'Download complete',
+    },
+    icons: {
+      default: <Icon src={Download} />,
+      pending: <Icon src={SpinnerSimple} className="icon-spin" />,
+      complete: <Icon src={Check} />,
+    },
+    disabledStates: ['pending', 'complete'],
+    variant: 'primary',
+  };
+
+  useEffect(() => {
+    if (!isPublic) {
+      const handleProgramUrlCopy = async () => {
```

Comment:
is there a reason we need to wrap this promise in an `async/await`? why not just have the first promise resolve inside `useEffect`?

Reply:
I see what you mean. Because the functions from the service file are already resolving promises, we don't need to wrap them in another `async/await`.